### ISSUE TITLE
twind/shim/server does not work with Node native ESM

### ISSUE THREAD

**Issue by rschristian** [tw-in-js/twind#183]
## Summary 

`twind/shim/server` doesn't work with Node native ESM, and from a quick glance through the build files, I expect most other exports won't work either in such an environment.

```
"./shim/server": {
  "esnext": "./shim/server/server.esnext.js",
  "module": "./shim/server/server.js",
  "types": "./shim/server/server.d.ts",
  "require": "./shim/server/server.cjs",
  "node": "./shim/server/server.mjs",
  "default": "./shim/server/server.js"
},
```

`module` in `exports` is not recognized by Node (and as far as I know, only Webpack supports it). Node therefore gets the `./shim/server/server.mjs` which doesn't match the same default / named exports.

`./shim/server/server.mjs`
```
import __$$ from "./server.cjs";
export const { shim } = __$$;
export default __$$;
```

Can't use `import { virtualSheet, getStyleTag, shim } from 'twind/shim/server'` with that. 

## Reproduce
Copy paste [this example from Twind's docs](https://twind.dev/handbook/the-shim.html#server) and run it with Node in a package that has `"type": "module"` set. 

#### Misc
The `.mjs` should be able to be an exact copy of the `.js` output in this case. Not sure if it's a bug or something else that would lead to this vastly different output.

**Comment by sastan** [comment]
I think I found the problem.

This works:

```js
import { shim } from 'twind/shim/server'

console.log(shim)
```

This does not because the build did not detect the [star exports](https://github.com/tw-in-js/twind/blob/main/src/shim/server/index.ts#L22-L23):

```js
import { getStyleTag } from 'twind/shim/server'

console.log(getStyleTag)
```

I try to find a solution.

---

`.mjs` pointing to `.cjs` is for the [recommended way](https://nodejs.org/dist/latest-v16.x/docs/api/packages.html#packages_approach_1_use_an_es_module_wrapper) for dual packages that stores internal state like twind does.

**Comment by sastan** [comment]
This should be fixed in v0.16.16.

**Comment by rschristian** [comment]
Fair enough on the `.mjs` pointing to `.cjs`, hadn't seen that.

Thanks for the super quick fix, works great! Really sets Twind above all others with the speed in which you fix any issues
